### PR TITLE
ROX-12743: Provide fleet-manager endpoint for manually deleting tenants in the database

### PR DESCRIPTION
**Pull request by mtesseract**
## Description

See https://issues.redhat.com/browse/ROX-12743. This PR implements a new DELETE endpoint as `/api/rhacs/v1/admin/dinosaurs/db/{CENTRAL ID}`.

## Checklist (Definition of Done)

- [x] Unit and integration tests added
- [x] ~~Documentation added if necessary~~
- [ ] CI and all relevant tests are passing
- [x] ~~Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~~

## Test manual

TODO: Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

File: internal/dinosaur/pkg/services/dinosaur.go [hunk 1]
```diff
@@ -517,30 +517,43 @@ func (k *dinosaurService) DeprovisionExpiredDinosaurs(dinosaurAgeInHours int) *e
...
 			}
+			glog.Infof("Successfully deleted CNAME records for Central tenant %q", centralRequest.ID)
 		}
 	}
 
 	// soft delete the dinosaur request
-	if err := dbConn.Delete(dinosaurRequest).Error; err != nil {
-		return errors.NewWithCause(errors.ErrorGeneral, err, "unable to delete dinosaur request with id %s", dinosaurRequest.ID)
+	if err := dbConn.Delete(centralRequest).Error; err != nil {
+		return errors.NewWithCause(errors.ErrorGeneral, err, "unable to delete dinosaur request with id %s", centralRequest.ID)
```
Comment (johannes94) [inline internal/dinosaur/pkg/services/dinosaur.go:550]: nit: change msg dinosaur to central, but maybe this will already be done in #434

**Review by johannes94 — APPROVED** [review]
LGTM

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -430,6 +431,80 @@ var _ = Describe("Central", func() {
...
+
+		// (1) Delete the Central CR.
+		centralRef := &v1alpha1.Central{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      centralName,
+				Namespace: namespaceName,
+			},
+		}
+		err = k8sClient.Delete(context.Background(), centralRef)
+		Expect(err).ToNot(HaveOccurred())
```
Comment (johannes94) [inline e2e/e2e_test.go:482]: FYI: The Expect clauses in a Describe Block lead to panics in CI execution I would suggest to put the cleanup in a ginkgo DeferCleanup container: https://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup

File: internal/dinosaur/pkg/services/dinosaur.go (hunk 1 above)
Comment (mtesseract) [reply internal/dinosaur/pkg/services/dinosaur.go:550]: Yes, looks like it will be done as part of the referenced PR.

File: e2e/e2e_test.go (hunk 2 above)
Comment (mtesseract) [reply e2e/e2e_test.go]: thanks, the code is changed.

File: e2e/e2e_test.go [hunk 3]
```diff
@@ -430,6 +429,87 @@ var _ = Describe("Central", func() {
...
+
+		It("should transition central's state to ready", func() {
+			Eventually(func() string {
+				return centralStatus(createdCentral, client)
+			}).WithTimeout(waitTimeout).WithPolling(defaultPolling).Should(Equal(constants.CentralRequestStatusReady.String()))
+		})
+
+		It("should be deletable in the control-plane database", func() {
+			err = adminClient.DbDeleteCentral(createdCentral.Id)
+			Expect(err).To(Succeed())
```
Comment (johannes94) [inline e2e/e2e_test.go:465]: I personally prefer:
```suggestion
			Expect(err).ToNot(HaveOccurred())
```
The intention of gomega Expects is to be read somewhat similar to natural language and `Expect(err).To(Succeed())` is not really clear to me.

**Review by johannes94 — APPROVED** [review]
LGTM

**Pull request merged**